_transcripts/032-rust-1.48-1.49.md

```diff
@@ -0,0 +1,731 @@
+And this wouldn't really work without const generics. Or it would be weird
+because it would only work for vectors of very small lengths. But now we can say
+this implementation, this method exists for any `N`, and I think this is really
+neat. I think this is a cool use of const generics.
+
+__Ben__: Yeah, it's really cool because previously would be kind of, like,
+awkward and difficult to make this conversion, you'd have to— you'd probably
+want to, like, initialize the array first and then manually copy, loop over all
+the fields, and then over each index, copy into it. And more likely, you would
+probably just use a slice. And so, in this case, we're going to start seeing, I
+believe, a lot of APIs that start actually getting to use these fixed-size
+arrays, which might not always be the most ergonomic or best way of doing
+things. But I'm sure there are plenty of, like, places where you don't wanna use
+a slice, you'd rather just use the fixed-size array.
+
+__Jon__: Yeah, we got into, when we were discussing this before we started, we
+got into, like, what other kind of interesting things might be coming down the
+line. You want to talk a little bit about the slice functions you found?
+
+__Ben__: Yeah. Let me pull those up real quick. I believe they were called,
+`slice::array_chunks`. Is that correct?
+
+__Jon__: Yeah. It's like, `as_chunks` and `array_chunks`.
+
+__Ben__: There's also an `array_windows` function somewhere there too? Let's
+bring it up real quick.
+
+__Jon__: Yeah, there's, like, `as_chunks_mut`, `array_chunks_mut`,
+`array_windows`. And all of these are— they're pretty neat because they're
+basically a way to say, because I happen to have them open here. If you have a
+slice and you want to get, sort of, sub-slices of that slice, of different
+lengths. Like, you have a slice of length 8, and you want every— you want, like,
+slices of 2, chunks of 2, into that slice. There's now in nightly, there's this
+new method on slices called `as_chunks`, where it's generic over the length of
+the slice, and it returns you, like, sliced arrays, that each is one of those
+chunks. So if you want to slice into chunks of 2, what you're going to get back
+is a slice of array slices of length 2, that are those chunks. And it's just
+really neat, I think.
+
+__Ben__: Yeah, I think it's one of those really cool things that, once we start
+seeing it in the wild, and start internalizing the use of these const generics,
+it's going to be, I think, like, pretty idiomatic, and pretty widely
+distributed. So let's say, get used to it because I believe 1.51 is when
+`min_const_generics` is going to finally make it into stable. I believe it was
+stabilized a few days ago, as of this recording. So we should be having a lot of
+fun in the future.
+
+And we should probably talk about (_unintelligible— 10:41_) some crates like the
+vector math crates which I believe really want to use these to their full
+potential.
```

Comment:
If you have good ears, feel free to try to figure out what was said at 10:41.

Reply:
"and what was that like" ? 😅

Reply:
I listened to it about 20 more times and I think it's maybe "And we should probably talk about, once that lands, ..."

which does make sense in context.